sierra_adapter/sierra_item_merger/src/main/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterService.scala

```diff
@@ -32,7 +33,26 @@ class SierraItemMergerUpdaterService @Inject()(
         }
     }
 
-    Future.sequence(updateFutures).map(_ => ())
+    val unlinkUpdateFutures = itemRecord.unlinkedBibIds.map { unlinkedBibId =>
+      mergedSierraRecordDao
+        .getRecord(unlinkedBibId)
+        .flatMap {
+          case Some(record) =>
+            val mergedRecord = record.unlinkItemRecord(itemRecord)
+            if (mergedRecord != record)
+              mergedSierraRecordDao.updateRecord(mergedRecord)
+            else Future.successful(())
+          // In the case we cannot find the bib record
+          // assume we're too early and put the message back
+          case None => Future.failed(SQSReaderGracefulException(
+            new RuntimeException("Missing Bib record to unlink: $unlinkedBibId.")
+          ))
+        }
```

Comment:
I think there might be some way of generalise this block inside the map. The only difference with the block in lines 20-33 is the call to `record.unlinkItemRecord(itemRecord)` instead of `record.mergeItemRecord(itemRecord)`.

Something like 
```scala
private def combineMergedRecordWithItem(bibId: String, itemRecord: SierraItemRecord)(
    f: MergedSierraRecord => SierraItemRecord =>  MergedSierraRecord) = {
    mergedSierraRecordDao
      .getRecord(bibId)
      .flatMap {
        case Some(record) =>
          val mergedRecord = f(record)(itemRecord)
          if (mergedRecord != record)
            mergedSierraRecordDao.updateRecord(mergedRecord)
          else Future.successful(())
        case None =>
          mergedSierraRecordDao.updateRecord(
            MergedSierraRecord(bibId,
                               itemData = Map(itemRecord.id -> itemRecord)))
      }
  }
```

that then gets called as 
```scala
val mergeUpdateFutures = itemRecord.bibIds.map { bibId =>
      combineMergedRecordWithItem(bibId, itemRecord)(
        _.mergeItemRecord)
    }
val unlinkUpdateFutures = itemRecord.bibIds.map { bibId =>
      combineMergedRecordWithItem(bibId, itemRecord)(
        _. unlinkItemRecord)
    }
```

I'm not sure if it's more readable but it definitely reduces code duplication

Reply:
👍  will clean up a bit in another PR